docs/components/Video/Video.js

```diff
@@ -10,26 +11,68 @@ export const Video = ({
   height = 315,
 }) => {
   const videoId = getUrlParamsFromString(url).find(({ key }) => key === 'v')?.value;
+  const [isVideoVisible, setIsVideoVisible] = useState(false);
+  const videoContainerRef = useRef();
+
+  useEffect(() => {
+    const options = {
+      root: null,
+      rootMargin: '0px',
+      threshold: 0.5, // Adjust this threshold as needed
+    };
+
+    let observer;
+    
+    if (videoContainerRef.current) {
+      observer = new IntersectionObserver(handleIntersection, options);
+      observer.observe(videoContainerRef.current);
+    }
+
+    return () => {
+      if (observer && videoContainerRef.current) {
+        observer.unobserve(videoContainerRef.current);
+      }
+    };
+  }, []);
+
+  const handleIntersection = (entries) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting) {
+        setIsVideoVisible(true);
+      }
+    });
+  };
 
   return (
     <figure className={styles.video}>
-      <div
-        className={styles.videoContainer}
-        style={{
-          paddingTop: `${(height / width) * 100}%`,
-        }}
-      >
-        <iframe
-          title={title}
-          width={width}
-          height={height}
-          src={`https://www.youtube.com/embed/${videoId}?feature=oembed`}
-          frameBorder="0"
-          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
-          allowFullScreen
-          autoPlay
-        />
-      </div>
+      {isVideoVisible ? (
+        <div
+          ref={videoContainerRef}
+          className={styles.videoContainer}
+          style={{
+            paddingTop: `${(height / width) * 100}%`,
+          }}
+        >
+          <iframe
+            title={title}
+            width={width}
+            height={height}
+            src={`https://www.youtube.com/embed/${videoId}?feature=oembed`}
+            frameBorder="0"
+            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+            allowFullScreen
+            autoPlay
+          />
+        </div>
+      ) : (
+        <div
+          ref={videoContainerRef}
+          style={{
+            width: '100%',
```

Comment:
i would think it would probably make sense for this to be the same value as the `isVideoVisible` div as we want it to take the same amount of space both horizontally and vertically as the real video

```
paddingTop: `${(height / width) * 100}%`,
```